### DISCUSSION TITLE
Sling terminate without error while source connection is killed

### DISCUSSION THREAD

**Discussion by EDORR-NEWORCH** [slingdata-io/sling-cli#407] [Q&A] [answered]
Hello,

I write a parquet file locally from a SQL Server data source. When I kill the SQL Source query, Sling stops processing but indicates that the task was completed successfully. 

This means that we ultimately have no way of being sure that we have recovered all the data in the parquet file. Is there a way to raise an error when the source is interrupted?

**Comment by flarco** [comment]
Hi, to clarify, you are not interrupting the sling process (with Ctrl+C for example), but killing the SQL query from inside SQL Server, correct?

**Reply by sdonati-orch** [reply]
Hi, this is correct.
We discoverd this during a network outage => stream stopped reading and as soon as the network went back data has been transferred without error (and number of rows transferred was below what we expected).
Killing the query was the way to repoduce the issue.

**Reply by EDORR-NEWORCH** [reply]
Hi @flarco ,

Do you plan to add error handling to resolve this issue ? If yes, what will be the deadline ?

Thanks in advance!!

**Reply by flarco** [reply]
Hey, I am not sure if it's possible. Sling uses the [Microsoft official driver](https://github.com/microsoft/go-mssqldb), and it should error by itself (like postgres driver, which errors if query is cancelled by server). I have another idea, but needs to be thought out a bit more.

**Reply by sdonati-orch** [reply]
Hi @flarco ,
His there any way you can help on this point?

**Reply by flarco** [reply]
hey @sdonati-orch I need to reproduce this on my side. 
Can you give me some more context? Is the query being killed intentionally? Or is the network connection between the machine running `sling` and the database server unstable? 
Also, can run with env var `DEBUG=TRACE` and give me the output? Please ensure to remove any sensitive logs. You could email the log output to `support@slingdata.io` if that works better for you.

**Comment by sdonati-orch** [comment]
Hey @flarco ,

Here is the difference between sqserver and mariadb executions with netwwork issue
=> in both case it ends with successful status even if we did not retrieve all the expected data (but with mariadb we can see network interruption)

![slung_run](https://github.com/user-attachments/assets/830dbe28-3ba0-45a8-8a9d-0866c2804778)

**Reply by flarco** [reply, marked as answer]
thanks @sdonati-orch @EDORR-NEWORCH, this was indeed a bug. I pushed a fix. Can you try with the latest [dev build](https://github.com/slingdata-io/sling-cli?tab=readme-ov-file#automated-dev-builds)?

**Reply by sdonati-orch** [reply]
Hi @flarco,

Looks good 😊

![image](https://github.com/user-attachments/assets/42c1a9ff-5ac4-44d7-91ea-307fbf4491ce)

Thanks a lot !!!!

**Comment by EDORR-NEWORCH** [comment]
I confirm, it's perfect!!!
Thanks a lot @flarco !!
